bin/proving-service/src/commands/proxy.rs

```diff
@@ -45,10 +45,26 @@ impl StartProxy {
     /// - The server cannot be started.
     #[tracing::instrument(target = MIDEN_PROVING_SERVICE, name = "proxy:execute")]
     pub async fn execute(&self) -> Result<(), String> {
+        // Check if all required ports are available
+        check_port_availability(&self.proxy_config.host, self.proxy_config.port, "Proxy")?;
+        check_port_availability(
+            &self.proxy_config.host,
+            self.proxy_config.workers_update_port,
+            "Workers update",
+        )?;
+
+        if self.proxy_config.metrics_config.enable_metrics {
+            check_port_availability(
+                &self.proxy_config.metrics_config.prometheus_host,
+                self.proxy_config.metrics_config.prometheus_port,
+                "Prometheus metrics server",
+            )?;
+        }
```

Comment:
Is it possible to pass in a `TcpListener` to the server? This improves the situation but technically there is still a gap where the ports can be bound by another process between the check and the server startup.

Reply:
Sadly, I did not find a way to pass listeners, it receives only `&str` in the related methods.